src/Modules/UngradedModules.jl

```diff
@@ -3515,39 +3523,52 @@ function +(a::SubquoModuleElem, b::SubquoModuleElem)
   check_parent(a,b)
   return SubquoModuleElem(coordinates(a)+coordinates(b), a.parent)
 end
+
 function -(a::SubquoModuleElem, b::SubquoModuleElem) 
   check_parent(a,b)
   return SubquoModuleElem(coordinates(a)-coordinates(b), a.parent)
 end
+
 -(a::SubquoModuleElem) = SubquoModuleElem(-coordinates(a), a.parent)
+
 function *(a::MPolyDecRingElem, b::SubquoModuleElem) 
   if parent(a) !== base_ring(parent(b))
     return base_ring(parent(b))(a)*b # this will throw if conversion is not possible
   end
   return SubquoModuleElem(a*coordinates(b), b.parent)
 end
+
 function *(a::MPolyRingElem, b::SubquoModuleElem) 
   if parent(a) !== base_ring(parent(b))
     return base_ring(parent(b))(a)*b # this will throw if conversion is not possible
   end
   return SubquoModuleElem(a*coordinates(b), b.parent)
 end
+
 function *(a::RingElem, b::SubquoModuleElem) 
   if parent(a) !== base_ring(parent(b))
     return base_ring(parent(b))(a)*b # this will throw if conversion is not possible
   end
   return SubquoModuleElem(a*coordinates(b), b.parent)
 end
+
 *(a::Int, b::SubquoModuleElem) = SubquoModuleElem(a*coordinates(b), b.parent)
 *(a::Integer, b::SubquoModuleElem) = SubquoModuleElem(a*coordinates(b), b.parent)
 *(a::QQFieldElem, b::SubquoModuleElem) = SubquoModuleElem(a*coordinates(b), b.parent)
+
 function (==)(a::SubquoModuleElem, b::SubquoModuleElem) 
   if parent(a) !== parent(b)
     return false
   end
   return iszero(a-b)
 end
 
+function Base.hash(a::SubquoModuleElem, h::UInt)
+  b = 0x0361a2f6467e4200 % UInt
+  h = hash(parent(a), h)
+  return xor(h, b)
```

Comment:
So this is (one of?) the contentious parts: we (@fieker and me) don't think this is really better (in practical applications) than a function returning a constant, and that of course is mostly useless. So we'd favor an error indicating "not implemented".

Actually, to make it stronger: we require this to merge the PR. However, of course if we run into actual problems caused by this, we can revisit the situation.

Reply:
Changed in 722eccf78db48580e5131a6a178f5555a9f80764.

I still think this is a bad idea and could potentially render Oscar useless for some users that rely on some julia Base functionality that uses hashing. I would prefer to return 0 or some constant and give a warning (that in the best case should make CI fail) instead. Then, some user can still use Oscar types for their use-case, although it may be very inefficient in those cases.

But if this is what you want, let's go with it.